scripts/Units/armguard.bos

```diff
@@ -100,45 +101,51 @@ aimCommon(heading, pitch)
 
 AimPrimary(heading, pitch)
 {
-	if (aimingState == 2){
+	if (aimCallCounter > aimFailCountThreshold){
+		aimCallCounter = 0;
+		aimingState = aimingState_forceHighAim;
 		return (0);
 	}
-	aimingState = 1;
-	start-script resetState();
+	if (aimingState > aimingState_aimingLow){
+		return (0);
+	}
+	aimingState = aimingState_aimingLow;
+	aimCallCounter = aimCallCounter + 1;
 	signal SIGNAL_AIM;
 	set-signal-mask SIGNAL_AIM;
+	start-script resetState();
 	call-script aimCommon(heading, pitch);
-
-	// Prevent low-trajectory "bonus shots" while high-trajectory is reloading
-	var reloading;
-	reloading = (get GAME_FRAME) < (get WEAPON_RELOADSTATE(2));
-	if( reloading )
-	{
-		return(0);
-	}
 	return (1);
 }
 
 AimSecondary(heading, pitch)
 {
 	
-	if (aimingState == 1){
+	if (aimingState == aimingState_aimingLow){
 		return (0);
 	}
 	var frame;
 	frame = (get GAME_FRAME);
-	if (aimingState == 2 && frame > resetHighFrame){
-		aimingState = 0;
+	if (aimingState == aimingState_aimingHigh && frame > resetHighFrame){
+		aimingState = aimingState_notAiming;
 		return (0);
 	}
-	if (aimingState != 2){
-		resetHighFrame = frame + delayFrames;
+	if (aimingState != aimingState_aimingHigh){
+		resetHighFrame = frame + (delayFrames * aimingState);
```

Comment:
Doing math on enum values is a code smell. Those numbers are arbitrary ID numbers, not any quantitative value.

Reply:
this is ancient, I'm sorry brother.